app/assets/javascripts/users.js.erb

```diff
@@ -146,8 +169,39 @@ function removeErrorMsgs() {
   $('.error-msg').remove();
 }
 
+function deselectTarget() {
+  if(selectedTarget != null) {
+    selectedTarget.setOptions({ fillColor: targetColor });
+    selectedTarget = null;
+    selectedTargetMarker = null;
+  }
+}
+
+function targetInfo(targetId) {
+  var route = $('.get-targets-path').data('path') + '/' + targetId;
```

Comment:
It's better to send the full destroy path from the server when loading the markers, that way you don't have to bother managing path on this side.